src/compiler/crystal/tools/doc/type.cr

```diff
@@ -786,49 +786,49 @@ class Crystal::Doc::Type
       builder.field "full_name", full_name
       builder.field "name", name
       builder.field "abstract", abstract?
-      builder.field "superclass" { (s = superclass) ? s.to_json_simple(builder) : builder.null }
+      builder.field "superclass" { superclass.try &.to_json_simple(builder) } unless superclass.nil?
       builder.field "ancestors" do
         builder.array do
           ancestors.each &.to_json_simple(builder)
         end
-      end
+      end unless ancestors.empty?
```

Comment:
Stylistic suggestion: Trailing conditionals are fine for single-line expressions, but when it spans multiple lines, a conditional block would be less confusing.